crates/flux-fhir-analysis/src/lib.rs

```diff
@@ -357,6 +359,25 @@ fn generics_of(genv: GlobalEnv, def_id: LocalDefId) -> QueryResult<rty::Generics
                 .ok_or_else(|| query_bug!(def_id.local_id(), "no generics for {def_id:?}"))?;
             conv::conv_generics(genv, generics, def_id, is_trait)
         }
+        DefKind::Ctor(..) => {
+            match def_id {
+                MaybeExternId::Local(local_id) => {
+                    // for a real local, get generics from parent
+                    let tcx = genv.tcx();
+                    let hir_id = tcx.local_def_id_to_hir_id(local_id);
+                    let parent_id = tcx.hir().get_parent_item(hir_id);
+                    let generics = genv.map().get_generics(parent_id.def_id)?.ok_or_else(|| {
+                        query_bug!(def_id.local_id(), "(ctor) no generics for {def_id:?}")
+                    })?;
+                    conv::conv_generics(genv, generics, def_id, false)
+                }
+                MaybeExternId::Extern(_, def_id) => {
+                    // for an extern, get generics from rustc
+                    let rustc_generics = genv.lower_generics_of(def_id);
+                    refining::refine_generics(genv, def_id, &rustc_generics)
+                }
+            }
+        }
```

Comment:
I think we can merge this into the arm below because the ctor is represented as a child of the `Enum` from the perspective of generics so the `rty::Generics` will always be empty and point to the parent, i.e., it'll look like

```
Generics { parent: Some(DefId(0:3 ~ playground[f6cf]::Foo)), parent_count: 1, own_params: [], param_def_id_to_index: {}, has_self: false, has_late_bound_regions: None }
```

Reply:
good catch!